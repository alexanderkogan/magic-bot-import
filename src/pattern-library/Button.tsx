import * as React from 'react'
import { Props } from '../react-typings'

export interface ButtonProps {
    clicked: () => void
}

export default (props: Props<ButtonProps>) => {
    return (
        <button type="button" onClick={props.clicked}>
            {props.children}
        </button>
    )
}
