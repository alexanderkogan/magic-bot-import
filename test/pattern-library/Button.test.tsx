import * as React from 'react'

import Button from '../../src/pattern-library/Button'
import { mount } from '../test-utils'

describe('Button', () => {
    it('renders', () => {
        const element = mount(<Button clicked={jest.fn()} />)
        expect(element).toMatchSnapshot()
    })

    it('calls callback on click', () => {
        const clicked = jest.fn()
        const element = mount(<Button clicked={clicked} />)
        element
            .find('button')
            .at(0)
            .simulate('click')
        expect(clicked).toHaveBeenCalled()
    })
})
