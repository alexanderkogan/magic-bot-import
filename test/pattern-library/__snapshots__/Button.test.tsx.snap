// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `
<Component
  clicked={[MockFunction]}
>
  <button
    onClick={[MockFunction]}
    type="button"
  />
</Component>
`;
